core/src/services/azfile/backend.rs

```diff
@@ -0,0 +1,426 @@
+
+        let config_loader = AzureStorageConfig {
+            account_name: self
+                .account_name
+                .clone()
+                .or_else(|| infer_account_name_from_endpoint(endpoint.as_str())),
+            account_key: self.account_key.clone(),
+            sas_token: self.sas_token.clone(),
+            ..Default::default()
+        };
+
+        let cred_loader = AzureStorageLoader::new(config_loader);
+
+        let signer = AzureStorageSigner::new();
+
+        debug!("backend build finished: {:?}", &self);
+        Ok(AzfileBackend {
+            core: Arc::new(AzfileCore {
+                root,
+                endpoint,
+                loader: cred_loader,
+                client,
+                signer,
+                share_name: self.share_name.clone(),
+            }),
+        })
+    }
+}
+
+fn infer_account_name_from_endpoint(endpoint: &str) -> Option<String> {
+    let endpoint: &str = endpoint
+        .strip_prefix("http://")
+        .or_else(|| endpoint.strip_prefix("https://"))
+        .unwrap_or(endpoint);
+
+    let mut parts = endpoint.splitn(2, '.');
+    let account_name = parts.next();
+    let endpoint_suffix = parts
+        .next()
+        .unwrap_or_default()
+        .trim_end_matches('/')
+        .to_lowercase();
+
+    if endpoint_suffix == DEFAULT_AZFILE_ENDPOINT {
+        account_name.map(|s| s.to_string())
+    } else {
+        None
+    }
+}
+
+/// Backend for azfile services.
+#[derive(Debug, Clone)]
+pub struct AzfileBackend {
+    core: Arc<AzfileCore>,
+}
+
+#[async_trait]
+impl Accessor for AzfileBackend {
+    type Reader = IncomingAsyncBody;
+    type BlockingReader = ();
+    type Writer = AzfileWriters;
+    type BlockingWriter = ();
+    type Pager = AzfilePager;
+    type BlockingPager = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::Azfile)
+            .set_root(&self.core.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                read: true,
+                read_can_next: true,
+                read_with_range: true,
+
+                write: true,
+                create_dir: true,
+                delete: true,
+                rename: true,
+
+                list: true,
+                list_with_delimiter_slash: true,
+
+                ..Default::default()
+            });
+
+        am
+    }
+
+    async fn create_dir(&self, path: &str, _: OpCreateDir) -> Result<RpCreateDir> {
+        self.core.ensure_parent_dir_exists(path).await?;
+        let resp = self.core.azfile_create_dir(path).await?;
+        let status = resp.status();
+
+        match status {
+            StatusCode::CREATED => {
+                resp.into_body().consume().await?;
+                Ok(RpCreateDir::default())
+            }
+            _ => {
+                if resp
+                    .headers()
+                    .get("x-ms-error-code")
```

Comment:
I think we cannot just check status code because `409 Conflict` has two meaning:
1. If a directory by the same name is being deleted when Create Directory is called, the server returns status code 409 (Conflict)
2. If a directory or file with the same name already exists, the operation fails with status code 409 (Conflict).

ref: https://learn.microsoft.com/en-us/rest/api/storageservices/create-directory

Reply:
Understood. Would you like to include those comments in the code for better clarity of its function?

Reply:
Cool, will do